src/i18next/factoryI18nextState.js

```diff
@@ -64,34 +92,82 @@ function factoryI18nextState(options) {
         }
       })
     },
+    resolveNamespace: (filename) =>
+      resolveNamespace(filename).split(nsSeparator).pop(),
     resolvers: (filename) => {
-      const namespace = clearNamespace(filename)
-      const initialStateResolvers = factoryResolversI18next(
-        namespace,
-        state.get(),
-      )
-      const getter = readable(initialStateResolvers, (set) => {
-        getter.unsubscribe = state.subscribe((newState) => {
-          set(factoryResolversI18next(namespace, newState))
+      const i18nextNamespace = resolveNamespaceI18next(filename)
+      const namespace = resolveNamespace(filename)
+      /**
+       * why do we need to fetch language with namespace here?
+       * namespaces in i18next are defined with 2 aspects
+       * 1. the json filename name to async load, '/assets/locales/{{lng}}/{{ns}}.json'
+       * 1. a parent node in the bundle translation file, '/assets/locales/{{lng}}.json'
+       * 2. the prefix of key resolution 'common:toolbar.confirm.text'
+       */
+      let i18nState = state.get()
+      fetchLanguageWithNamespace(
+        i18nState.i18n,
+        i18nState.i18nAsyncStatus,
+        i18nState.locale,
+        i18nextNamespace || defaultNamespace,
+      ).catch(onFetchError)
```

Comment:
Fetching asynchronously the resource through resolvers ensure that usage of i18n namespaces (do not get confused with `svelte-i18n` namespaces) are loaded on demand.